appveyor_dashboard.cmake

```diff
@@ -6,7 +6,7 @@ set(PY_VERSION_PATCH 8) # Value should match the default set in CMakeLists.txt
 if(NOT "$ENV{PY_VERSION_PATCH}" STREQUAL "")
   set(PY_VERSION_PATCH "$ENV{PY_VERSION_PATCH}")
 endif()
-set(CTEST_BUILD_NAME "2.7.${PY_VERSION_PATCH}-VS-$ENV{PLATFORM}-$ENV{CONFIGURATION}_$ENV{APPVEYOR_REPO_BRANCH}_${commit}")
+set(CTEST_BUILD_NAME "2.7.${PY_VERSION_PATCH}-VS-$ENV{PLATFORM}-$ENV{CONFIGURATION}_$ENV{APPVEYOR_PULL_REQUEST_TITLE}_#$ENV{APPVEYOR_PULL_REQUEST_NUMBER}_${commit}")
```

Comment:
Using cmake -P with this content:

set(var_with_hash "this_has_a_#")
message(${var_with_hash})

verifies it should be OK.